### ISSUE TITLE
Should we drop gid-based variants of Connect in NEST 3?

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#702]
When we introduced "NewConnect", we kept old variants of the `Connect` call for backward compatibility. `Connect` thus can be called in many ways
```
  source target	 Connect  
  source target	 syn_model Connect  
  source target weight delay	 Connect  
  source target weight delay syn_model Connect  
  source target params	 Connect  
  source target params	 syn_model Connect  
   
  sources targets	 Connect  
  sources targets conn_rule	 Connect  
  sources targets conn_spec	 Connect  
  sources targets conn_rule syn_model  Connect  
  sources targets conn_spec syn_model  Connect  
  sources targets conn_rule syn_spec   Connect  
  sources targets conn_spec syn_spec   Connect  
```

I am wondering whether we should drop the first six, gid-based variants in NEST 3 and rather add a variant with numeric values for `weight` and `delay` for the GidCollection-based variant.

**Comment by jessica-mitchell** [comment]
NEST Open Developer meeting says yes, 19.02.18

**Comment by jougs** [comment]
I'm working on this in the [Topology cleanup branch](https://github.com/compneuronmbu/nest-simulator/tree/fix-481-SubnetFreeTopology).

**Comment by jougs** [comment]
The changes are now implemented in cb8160494016ccd971808eaf25b074c307efd453 and I'm thus closing this issue.
